### ISSUE TITLE
Binary cross entropy in multi-class classification

### ISSUE THREAD

**Issue by derEitel** [derEitel/nitorch#13]
Hey @jpalbrecht  why do we check here for binary loss functions?
https://github.com/derEitel/nitorch/blob/5cc4d7a6d010549c159f45921816d595f73a4dcd/nitorch/inference.py#L118
